File: programmers/Level1/기사단원의 무기.py
def solution(number, limit, power):
    def get_power(x):
        s = set()
        for i in range(1, int(x ** .5) + 1):
            if not x % i:
                s.add(i)
                s.add(x // i)
        return len(s)

    res = 0
    for y in range(1, number + 1):
        p = get_power(y)
        res += (power if p > limit else p)
    return res

File: programmers/Level1/기사단원의 무기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int get_prime(int number) {
    int ret = 1;
    for (int i=1;i<number/2+1;i++){
        if(number % i == 0)
            ret++;
    }
    return ret;
}

int solution(int number, int limit, int power) {
    int answer = 0;
    int attack;
    for(int i=1;i<number+1;i++) {
        attack = get_prime(i);
        if (attack > limit)
            answer += power;
        else
            answer += attack;
    }
    return answer;
}

